src/dnaio/_core.pyx

```diff
@@ -380,12 +384,56 @@ def record_names_match(header1: str, header2: str):
     fastq-dump tool (used for converting SRA files to FASTQ) appends '.1', '.2'
     and sometimes '.3' to paired-end reads if option -I is used.
     """
-    cdef:
-        str name1, name2
-
-    name1 = header1.split()[0]
-    name2 = header2.split()[0]
-    if name1 and name2 and name1[-1] in '123' and name2[-1] in '123':
-        name1 = name1[:-1]
-        name2 = name2[:-1]
-    return name1 == name2
+    if (PyUnicode_KIND(header1) != PyUnicode_1BYTE_KIND or
+        PyUnicode_KIND(header2) != PyUnicode_1BYTE_KIND):
+        raise ValueError("Strings are not valid latin-1 encoded data.")
+    # No encoding, we can compare the unicode data directly. Provided it is
+    # in 1-byte encoding, so we can find the spaces and tabs easily.
+    cdef char * header1chars = PyUnicode_1BYTE_DATA(header1)
+    cdef char * header2chars = PyUnicode_1BYTE_DATA(header2)
+    return record_name_bytes_match(header1chars, header2chars, len(header1), len(header2))
+
+
+cdef object record_name_bytes_match(char *header1chars, char* header2chars, Py_ssize_t header1length, Py_ssize_t header2length):
```

Comment:
yes. But for some reason I do not understand it leads to a much slower run in the benchmarks. It negates most of the speed benefit. I have no idea why. This might be due to alignment issues.

Reply:
Apparantly it does not matter in another context. Changed.